### ISSUE TITLE
Add `start` to `Note` constructor as an alternative to `delta`

### ISSUE THREAD

**Issue by pmcharrison** [music-computing/amads#46]
We agreed that `delta` could be relegated to internal use only, with users being encouraged to use absolute time attribute `Note.start` and `Note.end`. How does this work when people create `Note` objects? Currently the way this works is that users create `Note` objects specifying `delta` and then later call `parent.insert` on those notes. What would the equivalent be if users create `Note` objects specifying `start` instead? If I create a `Note` object with `start=4`, and then insert it into a `Measure` which also has `start=4`, does the `Note` end up with `delta=0` or `delta=4`? Presumably `start` should be interpreted as an absolute time, and so `delta` ends up being 0? Need to check that this doesn't raise any complications down the line.

**Comment by pmcharrison** [comment]
I think one such complication is working out how `insert` can know whether a `Note` was created in absolute time or relative time. Currently `Note(start=4)` and `Note(delta=4)` would create identical objects (with `parent=None` and `delta=4`) and so `insert` would not know that it is meant to interpret the former in absolute time.

**Comment by rbdannenberg** [comment]
Except delta has meaning as beat position within the measure, which is used to advantage at least in some machine learning algorithms I'm familiar with. Given that and your example, I think delta is more natural for Note construction. If users really want to avoid thinking in terms of position-within-measure, maybe they should construct a flattened score and we should provide a function that inserts measures. In that case, delta is still the right thing to pass to the constructor, but since the parent start is zero, the Note will have start=delta.

**Comment by pmcharrison** [comment]
I think my issues could be addressed if we made `parent` part of the `__init__` signature, so that we know from the outset what `delta` is referring to. Would you be happy with this? It would necessitate a bit of refactoring (and maybe I hit unforseen issues with this) but I think it should protect us from some of these difficulties (e.g. `Note.start` delivering incorrect values until `insert` is called.

**Comment by pmcharrison** [comment]
Hi @rbdannenberg, what do you think about this one (my proposal to make `parent` part of the `Note.__init__` signature, so that we always know what `delta` is relative to)? I would be happy to do the refactoring myself.

**Comment by rbdannenberg** [comment]
@pmcharrison, yes, it's on my list. Feel free to implement when you are tired of waiting on me :-) (Also, speeding up XML input and removing weak links to parents are on my list.)

**Comment by pmcharrison** [comment]
Oh, that's nice to hear! Sure, I'll probably get round to it at some point if you don't. I think that should be my penalty for arguing for changing things over!
